### ISSUE TITLE
[2018.1.0-rc7] Configure Defaults vs Domain Level Inheritance

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#330]
I have the following setup in `configure.yaml` for one of the questions:
```
  - &pri_network_interface
    identifier: pri_network_interface
    question: 'Interface for the primary network'
    default: 'eth0'
```

When I run through `metal configure domain` I set the interface to `em1`, which gives me the following in `answers/domain.yaml`:
```
pri_network_interface: em1
```

When running through `metal configure local` I see the domain-level answer as the one it would use:
```
Interface for the primary network (3/60)  |em1|
eth0
```

But if I change this to `eth0` (the default in `configure.yaml`) I get the following in `answers/nodes/local.yaml` (note how the interface isn't specified):
```
pri_network_defined: true
pri_network_ip: 10.10.0.1
```

This means that it ends up inheriting the domain answer instead of using the configure default, thus, meaning I can't use `eth0` unless it's specified at the domain level to match the default:
```
[root@localhost metalware]# metal each local 'echo <%= answer.pri_network_interface %>'
em1
```

This also effects the groups which I configure to use the domain default, the answers file will specify the interface name again, this could cause a lot of confusion with inheritance (below is `answers/groups/nodes.yaml`):
```
pri_network_defined: true
pri_network_interface: em1
```

**Comment by bobwhitelock** [comment]
@WilliamMcCumstie can you look into this? As I believe you implemented how we determined whether to save things based on what the default and higher answers are. Looks like we should check the higher level answer if that is present and use that to decide whether we should save the answer at the level being configured, rather than only checking the default.

You'll probably also want to look at my 2 open PRs first as they also touch the `Configurator`.

**Comment by ColonelPanicks** [comment]
In addition to this it looks like the answers are being saved at lower levels if they match the setting at a higher level. This can be a bit of nuisance when everything is using the `domain` answer (which differs from the `configure.yaml` default) but a later change to the `domain` answer isn't propagated.

**Comment by bobwhitelock** [comment]
@ColonelPanicks yep looks like the higher level answers aren't being considered at all when considering whether to save an answer, though they are picked up when determining the default to show. Ideally the whole answers hierarchy would be considered when determining whether to save an answer at a particular level, rather than just the `configure.yaml` default. Do you know if this has ever worked differently though, or has this just not been noticed until now?

**Comment by ColonelPanicks** [comment]
I don't believe this has ever worked, with the domain usually using the same defaults as `configure.yaml` and with very little reconfiguring being done it hasn't be noticed all too much
